docs/typescript/logging.md

```diff
@@ -13,51 +13,89 @@ Activities run in the standard Node.js environment and can use any Node.js logge
 <!--SNIPSTART typescript-logger-sink-interface-->
 <!--SNIPEND-->
 
-- Sinks are typically used for exporting logs, metrics and traces out from the Workflow into the isolate.
-- **Injected WorkflowInfo argument**: The first argument of a Sink Function implementation will be a [`workflowInfo` object](https://typescript.temporal.io/api/interfaces/workflow.workflowinfo/) that containing useful metadata.
-- **No return value**: Sink functions may not return values to the Workflow in order to prevent breaking determinism.
+### Implementing Sinks
 
-### Logger Sinks Example
+Implementing Sinks is a two step process.
 
-Call a Sink function from a Workflow
+#### Implement and inject the Sink function into a Worker
 
-<!--SNIPSTART typescript-logger-sink-workflow-->
+<!--SNIPSTART typescript-logger-sink-worker-->
 <!--SNIPEND-->
 
-Inject a function as a Workflow Sink
+- Sink function implementations are passed as an object into [WorkerOptions](https://typescript.temporal.io/api/interfaces/worker.workeroptions/#dependencies)
+- You may specify whether or not you'd like the injected function to be called during Workflow replay with the `callDuringReplay` boolean option.
 
-<!--SNIPSTART typescript-logger-sink-worker-->
+#### Proxy and call a Sink function from a Workflow
+
+<!--SNIPSTART typescript-logger-sink-workflow-->
 <!--SNIPEND-->
 
-### Sinks vs Activities
+Some important features of the [InjectedSinkFunction](https://typescript.temporal.io/api/interfaces/worker.InjectedSinkFunction) interface:
 
-Sinks are similar to Activities in that they are both registered on the Worker and proxied into the Workflow.
-However, they differ from Activities in important ways:
+- **Injected WorkflowInfo argument**: The first argument of a Sink function implementation will be a [`workflowInfo` object](https://typescript.temporal.io/api/interfaces/workflow.workflowinfo/) that containing useful metadata.
+- **Primitive arguments only**: The remaining Sink function arguments are copied between the isolated vm and the Node.js environment. This limits the usage to primitive types such as `number`, `string`, `array` and `object`.
+- **No return value**: Sink functions may not return values to the Workflow in order to prevent breaking determinism.
+
+#### Advanced: Performance Considerations and Non-blocking Sinks
+
+Be aware that Sinks will take up part of the workflow execution time.
 
-- Sinks Functions don't return promises
-- Sinks calls are not recorded in Workflow histories (no timeouts or retries)
-- Sink Functions are run on the same Worker that runs the Workflow they are called from.
+- If you have a long running sink function, e.g. one tries to communicate with external services, you may start seeing workflow task timeouts.
+- The effect of this is multiplied when replaying long workflow histories because the workflow task timer starts when the first history page is delivered to the worker and the sink function will run every time the workflow is "activated".
```

Comment:
```suggestion
- The effect is multiplied when replaying long Workflow histories because the Workflow Task timer starts when the first history page is delivered to the Worker, and the Sink function runs every time the workflow is "activated."
```